components/formats-bsd/src/loci/formats/in/JPEGReader.java

```diff
@@ -207,8 +212,47 @@ protected void initFile(String id) throws FormatException, IOException {
         }
 
         HashMap<String, String> tags = exif.getTags();
-        for (String tagName : tags.keySet()) {
-          addGlobalMeta(tagName, tags.get(tagName));
+        String unitsTagName = exif.getTagName(ExifSubIFDDirectory.TAG_RESOLUTION_UNIT);
+        String unitsTag = tags.get(unitsTagName);
+        Unit<Length> units = UNITS.MICROMETER;
+        if (unitsTag != null) {
+          if (unitsTag.toLowerCase().contains("inch")) { 
+            units = UNITS.INCH;
+          }
+          else if (unitsTag.toLowerCase().contains("centimeter")) { 
+            units = UNITS.CENTIMETER;
+          }
+          else if (Integer.parseInt(unitsTag) == 2) { 
+            units = UNITS.INCH;
+          }
+          else if (Integer.parseInt(unitsTag) == 3) { 
+            units = UNITS.CENTIMETER;
+          }
+        }
+        for (String tagName : tags.keySet()) {  
+          if (tagName.equals(exif.getTagName(ExifSubIFDDirectory.TAG_X_RESOLUTION))) {
+            try {
+              double resolutionX = Double.parseDouble(tags.get(tagName));
+              Length sizeX = FormatTools.getPhysicalSizeX((1/resolutionX),units);
+              store.setPixelsPhysicalSizeX(sizeX, 0);
+            }
+            catch(NumberFormatException e) {
+              LOGGER.warn("EXIF Tag - Resolution X not in correct format, expected double, found : " + tags.get(tagName));
```

Comment:
My main concern with the usage of `WARN` level is that it can be invasive e.g. in the Fiji console and/or OMERO logs. Alternatively we could use `DataTools.parseDouble()` which will log incorrect formatting at `DEBUG` level.